### PR TITLE
Feature/linter optimizer checks

### PR DESCRIPTION
**Pull request by tharders**
Implementing linter page experience checks for
* Page is transformed AMP
* Page is using Java Script Module version of the AMP Runtime.

### CODE REVIEW

**Comment by tharders** [comment]
I wonder if we should somehow add the recommendation to the linter rule (for example in the info field).
Otherwise it might not be clear what to do when running the tests in the command line.
(It looks like the "info" is currently not printed for results in the command line)

File: packages/linter/src/rules/IsTransformedAmp.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Context } from "../index";
+import { Rule } from "../rule";
+
+export class IsTransformedAmp extends Rule {
+  run({ $ }: Context) {
+    const isTransformed = $("html[transformed='self;v=1']").length > 0;
```
Comment (sebastianbenz) [inline packages/linter/src/rules/IsTransformedAmp.ts:6]: Checking for the presence of `transformed` is enough. The content might change.

File: packages/linter/src/rules/ModuleRuntimeUsed.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { Context } from "../index";
+import { Rule } from "../rule";
+
+export class ModuleRuntimeUsed extends Rule {
+  run({ $ }: Context) {
+    const isTransformed = $("html[transformed='self;v=1']").length > 0;
```
Comment (sebastianbenz) [inline packages/linter/src/rules/ModuleRuntimeUsed.ts:6]: Might make sense to extract this into a helper method. We'll need it more often.

File: packages/linter/src/rules/ModuleRuntimeUsed.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    if (!isTransformed) {
+      return this.pass();
+    }
+    const isModuleVersion =
+      $("script[type='module'][src='https://cdn.ampproject.org/v0.mjs']")
```
Comment (sebastianbenz) [inline packages/linter/src/rules/ModuleRuntimeUsed.ts:11]: Just check for `/v0.mjs` to support runtime self-hosting in the future.

File: packages/linter/src/rules/ModuleRuntimeUsed.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        .length > 0;
+    return isModuleVersion
+      ? this.pass()
+      : this.warn(
+          "The Java Script module version of the AMP Runtime is not used"
```
Comment (sebastianbenz) [inline packages/linter/src/rules/ModuleRuntimeUsed.ts:16]: s/Java Script/JavaScript

File: packages/linter/src/rules/IsTransformedAmp.ts (hunk 1 above)
Comment (tharders) [reply packages/linter/src/rules/IsTransformedAmp.ts]: done

File: packages/linter/src/rules/ModuleRuntimeUsed.ts (hunk 2 above)
Comment (tharders) [reply packages/linter/src/rules/ModuleRuntimeUsed.ts]: done

File: packages/linter/src/rules/ModuleRuntimeUsed.ts (hunk 3 above)
Comment (tharders) [reply packages/linter/src/rules/ModuleRuntimeUsed.ts]: done

File: packages/linter/src/rules/ModuleRuntimeUsed.ts (hunk 4 above)
Comment (tharders) [reply packages/linter/src/rules/ModuleRuntimeUsed.ts]: done

**Comment by sebastianbenz** [comment]
Build failure is unrelated

**Pull request merged**